pkg/scheduler/factory.go

```diff
@@ -528,3 +528,14 @@ func GetPodDisruptionBudgetLister(informerFactory informers.SharedInformerFactor
 	}
 	return nil
 }
+
+// v1DeepEqualSafePodSpec returns a PodSpec which is ready to be used with apiequality.Semantic.DeepEqual
```

Comment:
see https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/cli-runtime/pkg/resource/helper_test.go#L59
maybe we can use this v1DeepEqualSafePodSpec after moving it to helper.go not stay here.
/cc @damemi 
What do you think?

Reply:
copy here?

Reply:
Yeah that's a good question, because on one hand it's already an exported function there, but on the other I only see `cli-runtime` being imported by kubectl-related code. So I don't think it would really fit here.

However, does this even need to be a function? I see it's only being called once in `factory_test.go` and it returns a constant PodSpec. So could we just put this podspec in that spot instead?

Reply:
So could we just put this podspec in that spot instead?
Agree

Reply:
updated , pls review it again.

Reply:
/cc @damemi